substrate/pkg/controller/substrate/cluster/config.go

```diff
@@ -28,35 +28,43 @@ import (
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/service/sts"
+	iamauthenticatoraddon "github.com/awslabs/kit/operator/pkg/utils/iamauthenticator"
```

Comment:
I think we should avoid this dependency if at all possible. Else we tightly couple the dependency closure (and k8s version upgrade) of these projects.

Reply:
There is a lot of duplicate code like in this case which I was trying to reuse. I think long term we need to discuss if we want to extract out common code into a separate package